numpy/lib/recfunctions.py

```diff
@@ -987,7 +992,7 @@ def _unstructured_to_structured_dispatcher(arr, dtype=None, names=None,
     return (arr,)
 
 @array_function_dispatch(_unstructured_to_structured_dispatcher)
-def unstructured_to_structured(arr, dtype=None, names=None, align=False,
+def _unstructured_to_structured(arr, dtype=None, names=None, align=False,
```

Comment:
The private functions don't need the `array_function_dispatch` decorator, which means all the *_dispatcher functions for these are no longer needed as well. It would be nice to clean them up